src/vaadin-message-list.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { PolymerElement, html } from '@polymer/polymer/polymer-element.js';
+import '@polymer/polymer/lib/elements/dom-repeat.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+import { ElementMixin } from '@vaadin/vaadin-element-mixin/vaadin-element-mixin.js';
+import './vaadin-message.js';
+/**
+ * `<vaadin-message-list>` is a Web Component for showing an ordered list of messages. The messages are rendered as <vaadin-message>
+ *
+ * ```html
+ * <vaadin-message-list
+ *   items='[{"text":"Hello list","time":"yesterday", "user": {"name":"Matt Mambo","abbr":"MM","colorIndex":"1"}},{"text":"Hello list","time":"right now", "user": {"name":"Linsey Listy","abbr":"LL","colorIndex":"2"}}]'>
+ * </vaadin-message-list>
+ * ```
+ *
+ * ### Styling
+ *
+ * The following shadow DOM parts are available for styling:
+ *
+ * Part name | Description
+ * ----------|----------------
+ *
+ * See [ThemableMixin – how to apply styles for shadow parts](https://github.com/vaadin/vaadin-themable-mixin/wiki)
+ *
+ * @extends HTMLElement
+ * @mixes ThemableMixin
+ * @mixes ElementMixin
+ */
+class MessageListElement extends ElementMixin(ThemableMixin(PolymerElement)) {
+  static get properties() {
+    return {
+      /**
+       * A user object that can be used to render avatar and name.
+       * The user object can consist of the folowing properties:
+       * ```js
+       * Array<{
+       *   text: string,
+       *   time: string,
+       *   user: {
+       *     name: string,
+       *     abbr: string,
+       *     img: string,
+       *     colorIndex: number
+       *   }
+       * }>
+       * ```
+       */
+      items: {
+        type: Array,
+        value: function () {
+          return [];
+        }
+      }
+    };
+  }
+
+  static get template() {
+    return html`
+      <style>
+        :host {
+          display: block;
+          overflow: auto;
+        }
+
+        :host([hidden]) {
+          display: none !important;
+        }
+      </style>
+      <template is="dom-repeat" items="[[items]]">
+        <vaadin-message time="[[item.time]]" user="[[item.user]]">[[item.text]]</vaadin-message>
+      </template>
+    `;
+  }
+
+  ready() {
+    super.ready();
```

Comment:
Since `vaadin-message` now uses `this.setAttribute('role', 'listitem');` - this should be updated to include role=list

Reply:
Additionally: I'm not sure if it should be possible to use `vaadin-message` without `vaadin-message-list` because now the default `vaadin-message` violates the wai aria recommendation: 

> Authors **MUST** ensure elements whose role is listitem are contained in, or owned by, an element whose role is list.

Source: https://www.w3.org/TR/wai-aria-1.2/#listitem

Reply:
Hey @knoobie! Thanks for your input!

I created the base of both `vaadin-message` and `vaadin-message-list`, and actually neither had the wai-aria rules or the correct styling in place. @anezthes, our designer, is following my wake, fixing styling and ensuring wai aria compatibility. He fixed the `vaadin-message`company today in #13, and started looking at `vaadin-message-list` today. It might come in this PR or a separate one, but it is coming.

@anezthes, make sure that knoobies' comment is taken into account, and `vaadin-message-list` has role="list".

It is an interesting point that role="listitem" must be inside an container with role="list". Might be that we have to move the logic of adding role=listitem to `vaadin-message` into `vaadin-message-list`.

Reply:
@Peppe I created a separate issue with all my thoughts to reduce the noise in the initial pr - please take a look and comment if there is something totally wrong with my assumption or you have even better ideas for a great UX :) https://github.com/vaadin/vaadin-messages/issues/14

Reply:
@knoobie looks great! They sound sane to me, but I'm far away from a wai aria or screen reader expert. We will have a look at them!

Reply:
I added list="role" while I was fixing other things in the component. We will take care of the rest of the feedback in another PR.

Reply:
Done.